chunk_arrow.go

```diff
@@ -60,57 +52,39 @@ func (arc *arrowResultChunk) decodeArrowChunk(ctx context.Context, rowType []exe
 	}
 }
 
-/**
-Build arrow chunk based on RowSet of base64
-*/
-func buildFirstArrowChunk(rowsetBase64 string) arrowResultChunk {
-	rowSetBytes, err := base64.StdEncoding.DecodeString(rowsetBase64)
-	if err != nil {
-		return arrowResultChunk{}
-	}
-	rr, err := ipc.NewReader(bytes.NewReader(rowSetBytes))
-	if err != nil {
-		return arrowResultChunk{}
-	}
-
-	return arrowResultChunk{*rr, 0, 0, memory.NewGoAllocator()}
-}
-
-
-// Writes []array.Record to array record channel. Returns number of rows from records written to channel
-func (arc *arrowResultChunk) writeToArrowChan(ch chan<- []array.Record, rowType []execResponseRowType) (int, error) {
-	var numRows int
+func (arc *arrowResultChunk) decodeArrowBatch(scd *snowflakeChunkDownloader) (*[]array.Record, error) {
 	var records []array.Record
 
 	for {
 		rawRecord, err := arc.reader.Read()
 		if err == io.EOF {
-			ch <- records
-			return numRows, nil
+			break
 		} else if err != nil {
-			return numRows, err
+			return nil, err
 		}
-		record, err := arrowToRecord(rawRecord, rowType)
+		record, err := arrowToRecord(rawRecord, scd.RowSet.RowType)
 		rawRecord.Release()
 		if err != nil {
-			return numRows, err
+			return nil, err
 		}
+		record.Retain()
```

Comment:
isn't it strange to retain this?

Reply:
get a `malformed data` in the do_get otherwise :(